docs/types/reference-types.rst

```diff
@@ -37,6 +37,11 @@ non-persistent area where function arguments are stored, and behaves mostly like
     data location can also be returned from functions, but it is not possible to
     allocate such types.
 
+.. note::
+    Arrays and structs with ``calldata`` data location declared in a function body
+    or as its return parameters must be assigned before being used or returned.
+    This coding style is a workaround for allowing complex initialization patterns.
```

Comment:
This still is not correct.
The compiler enforces a requirement that calldata arrays and structs (declared as function local variables or return parameters) must be initialized before used.
Unfortunately, in certain cases, in which a complex control flow is used, the compiler can't detect the initialization.
A common workaround utilized in those cases is to assign the problematic variables to themselves to satisfy the rule enforced by the compiler.